### DISCUSSION TITLE
Сommon way to get the СhatId of any type message

### DISCUSSION THREAD

**Discussion by ffrip** [Eptagone/Telegram.BotAPI#53] [Q&A]
```c#
if (update.Type == UpdateType.Message)
    chatId = update.Message.Chat.Id;
```
but how handle all other types?
```c#
if (update.Type != UpdateType.Message)
{
    chatId = ???;
    await bot.SendMessageAsync(chatId, "Message is not a message-type");
}
```

**Comment by Eptagone** [comment]
Hi, it depends. In most cases you will have a copy of the message which you can use to obtain the chat id
For example, you can get the message from a callback query>
```CSharp
if (update.Type == UpdateType.CallbackQuery)
{
    var chatId = update.CallbackQuery.Message?.Chat.Id;
    if (chatId != null)
    {
        await bot.SendMessage(chatId, "Yeah");
    }
    await bot.AnswerCallbackQueryAsync(update.CallbackQuery.Id);
}
```
But be careful, the message may not always be available.
Try taking a look at the [official documentation](https://core.telegram.org/bots/api#update) to see what properties you could use to get the Chat Id.
